smart-contracts/contracts/dex-router-osmosis/src/error.rs

```diff
@@ -31,6 +31,9 @@ pub enum ContractError {
 
     #[error("Can't set empty path.")]
     EmptyPath {},
+
+    #[error("Not enough liquidity for swap from {offer:?} to {ask_denom:?}")]
+    NotEnoughLiquidity { offer: Coin, ask_denom: String },
```

Comment:
Is that error only about liquidity issues? May it be for another reason so would make more sense a generic SwapError from asset to asset?

Reply:
done, gave the error a more generic name, other possible causes are
- osmosis messed up (though we shouldn't care about that in our implementation)
- a pool that is used in a path got removed